wallet/wallet-cli-lib/src/commands/mod.rs

```diff
@@ -231,382 +246,501 @@ fn print_coin_amount(chain_config: &ChainConfig, value: Amount) -> String {
-                wallet,
-            ));
-
-            let msg = if need_mnemonic_backup {
-                format!(
-                    "New wallet created successfully\nYour mnemonic: {}\nPlease write it somewhere safe to be able to restore your wallet."
-                , mnemonic)
-            } else {
-                "New wallet created successfully".to_owned()
-            };
-            Ok(ConsoleCommand::Print(msg))
-        }
+struct CLIWalletState {
+    selected_account: U31,
+    total_accounts: usize,
+}
 
-        WalletCommand::OpenWallet { wallet_path } => {
-            utils::ensure!(
-                controller_opt.is_none(),
-                WalletCliError::WalletFileAlreadyOpen
-            );
+pub struct CommandHandler {
+    // the CLIWalletState if there is a loaded wallet
+    state: Option<CLIWalletState>,
+}
 
-            let wallet = RpcController::open_wallet(Arc::clone(chain_config), wallet_path)
-                .map_err(WalletCliError::Controller)?;
+impl CommandHandler {
+    pub fn new() -> Self {
+        CommandHandler { state: None }
+    }
+
+    fn set_total_accounts(&mut self, new_total_accounts: usize) {
+        if let Some(CLIWalletState {
+            selected_account: _,
+            total_accounts,
+        }) = self.state.as_mut()
+        {
+            *total_accounts = new_total_accounts;
+        } else {
+            self.state.replace(CLIWalletState {
+                selected_account: DEFAULT_ACCOUNT_INDEX,
+                total_accounts: new_total_accounts,
+            });
+        }
+    }
 
-            *controller_opt = Some(RpcController::new(
-                Arc::clone(chain_config),
-                rpc_client.clone(),
-                wallet,
-            ));
+    fn set_selected_account(&mut self, account_index: U31) -> Result<(), WalletCliError> {
+        let CLIWalletState {
+            selected_account,
+            total_accounts,
+        } = self.state.as_mut().ok_or(WalletCliError::NoWallet)?;
 
-            Ok(ConsoleCommand::Print(
-                "Wallet loaded successfully".to_owned(),
-            ))
+        if account_index.into_u32() as usize >= *total_accounts {
+            return Err(WalletCliError::AccountNotFound(account_index));
         }
 
-        WalletCommand::CloseWallet => {
-            utils::ensure!(controller_opt.is_some(), WalletCliError::NoWallet);
+        *selected_account = account_index;
+        Ok(())
+    }
 
-            *controller_opt = None;
+    fn selected_account(&self) -> Option<U31> {
+        self.state.as_ref().map(|state| state.selected_account)
+    }
 
-            Ok(ConsoleCommand::Print("Success".to_owned()))
+    fn repl_status(&mut self) -> String {
+        match self.state {
+            Some(CLIWalletState {
+                selected_account,
+                total_accounts,
+            }) if total_accounts > 1 => {
+                format!("({}/{})", selected_account, total_accounts)
+            }
```

Comment:
I suspect `selected_account` start indexing at 0. It may be a confusing UI when using the `(X/N)` notation, since `X`  does not really reach `N`, i.e. users may expect the last account to be called `N` rather than `N-1`. Not that I have a good idea how to resolve it though.

Reply:
yes exactly, it shows 0/2 and 1/2, but not sure what is the best solution to this.